src/validation.cpp

```diff
@@ -1936,6 +1937,35 @@ static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 static int64_t nBlocksTotal = 0;
 
+bool CheckPeginRipeness(const CBlock& block, const std::vector<std::pair<CScript, CScript>>& fedpegscripts) {
+    for (unsigned int i = 0; i < block.vtx.size(); i++) {
+        const CTransaction &tx = *(block.vtx[i]);
+
+        if (!tx.IsCoinBase()) {
+            for (unsigned int i = 0; i < tx.vin.size(); ++i) {
+                if (tx.vin[i].m_is_pegin) {
+                    std::string err;
+                    // IsValidPeginWitness technically checks more than just "ripeness". As a result,
+                    // if we get a validly-signed block which contains an ill-formed pegin witness,
+                    // this test will fail, and it will be treated as "unripe" rather than invalid,
+                    // resulting in a stall rather than rejecting the block. This is reasonable, because 
```

Comment:
trailing whitespace

Reply:
Also, I think this comment is incorrect because in the case that `IsValidPeginWitness` fails, we match on the error string.

Reply:
Ah, the comment is removed 2 commits later